src/Service/SF1500/AbstractDataFetcher.php

```diff
@@ -96,7 +98,9 @@ protected function logFetchProgress(int $total, int $max): void
      */
     protected function logFetchFinished(int $total): void
     {
-        $this->logger->debug(sprintf('Finished fetching %s data. Fetched a total of %d records.', static::DATA_TYPE, $total));
+        $this->logger->debug(sprintf('Finished fetching %s data.', static::DATA_TYPE));
+        $this->logger->debug(sprintf('Fetched a total of %d records.', $total));
+        $this->logger->debug(sprintf('Elapsed time: %s', gmdate('H:i:s', strtotime('now') - $this->start)));
```

Comment:
```suggestion
        $now = new \DateTimeImmutable();
        $this->logger->debug(sprintf('Elapsed time: %s', $now->diff($this->start)->format('%H:%I:%S'));
```

Consider reporting elapsed time in `logFetchProgress` as well.

Reply:
Used OOP and reported elapsed time in `logFetchProgress` as well as `logFetchFinished`.